goon-universe/queries.js

```diff
@@ -614,45 +689,25 @@ function getFinalReflectPage(scenarioID, callback){
 }
 
 //Returns question IDs as well for getChoices functions
-function getInitActionPageQuestions(scenarioID, callback){
-    let thisQuery= 'select question.question, question.id from pages, mcq, question where pages.id = mcq.page_id and mcq.page_id=question.mcq_id and pages.order ='+ INIT_ACTION +'and scenario_id = $1'
-    pool.query(thisQuery, [scenarioID], (error,results) => {
-        if (error) {
-            throw error
-        }
-        callback(results.rows)
-    })  
-}
-
-
-function getInitActionPageChoices(scenarioID, questionId, callback){
-    let thisQuery= 'select mcq_option.option from pages, mcq, mcq_option, question where mcq_option.question_id= $1 and mcq_option.question_id = question.id and pages.id = mcq.page_id and mcq.page_id=question.mcq_id and pages.order ='+ INIT_ACTION +'and scenario_id = $2'
-    pool.query(thisQuery, [questionId, scenarioID], (error,results) => {
-        if (error) {
-            throw error
-        }
-        callback(results.rows)
-    })  
-}
 
-function getFinalActionPageQuestions(scenarioID, callback){
-    let thisQuery= 'select question.question from pages, mcq, question where pages.id = mcq.page_id and mcq.page_id=question.mcq_id and pages.order ='+ FINAL_ACTION +'and scenario_id = $1'
-    pool.query(thisQuery, [scenarioID], (error,results) => {
-        if (error) {
+function getInitActionPageQuestionsAndChoices(scenarioID,questionID,callback){
+    let thisQuery='select question.question, question.id, mcq_option.option from pages, mcq, mcq_option, question where pages.order ='+ INIT_ACTION +'and scenario_id = $1 and mcq.page_id = question.mcq_id and pages.id=mcq.page_id and mcq_option.question_id = $2 and mcq_option.question_id = question.id'
+    pool.query(thisQuery, [scenarioID, questionID], (error,results) => {
+        if(error){
             throw error
         }
         callback(results.rows)
-    })  
+    })
 }
```

Comment:
Why is `getInitActionPageQuestionsAndChoices()` taking in a questionID parameter, but also returing question IDs?

Reply:
Fixed in 05936594.